src/client/power-pages/create/utils/MultiStepInput.ts

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+
+import { Disposable, QuickInput, QuickInputButton, QuickInputButtons, QuickPickItem, window } from "vscode";
+
+class InputFlowAction {
+    static back = new InputFlowAction();
+    static cancel = new InputFlowAction();
+    static resume = new InputFlowAction();
+}
+
+type InputStep = (input: MultiStepInput) => Thenable<InputStep | void>;
+
+interface QuickPickParameters<T extends QuickPickItem> {
+    title: string;
+    step: number;
+    totalSteps: number;
+    items: T[];
+    activeItem?: T;
+    placeholder: string;
+    buttons?: QuickInputButton[];
+}
+
+interface InputBoxParameters {
+    title: string;
+    step: number;
+    totalSteps: number;
+    value: string;
+    prompt?: string;
+    placeholder: string;
+    validate: (value: string) => Promise<string | undefined>;
+    buttons?: QuickInputButton[];
+}
+
+export class MultiStepInput {
+    static async run(start: InputStep) {
```

Comment:
Coding convention: 
In class we should declare all the member variables then all methods definition